### DISCUSSION TITLE
How do I add a search component? The official Documentation reads, "Documentation Coming soon..."

### DISCUSSION THREAD

**Discussion by sunlifeng212** [vuejs/vitepress#1154] [Q&A]
How do I add a search component? The official Documentation reads, "Documentation Coming soon..."

**Comment by brc-dd** [comment]
https://github.com/vuejs/vitepress/issues/1078

**Comment by MarkusKeck** [comment]
Adding the search component works with your attached issue.
Is there any documentation how algolia can scan your domain to automatically create the search items in combination with vitepress?
I read that this kind of crawler is not included in the free version of algolia, if that is true I doubt that this will find a huge following in the vitepress community. 

I also saw that in milestone for vitepress 1.0.0 there is one task attached for offline search. Sadly there is no date specified when this is going to happen.
https://github.com/vuejs/vitepress/milestone/1

**Comment by Charles7c** [comment]
DocSearch is a service that requires an additional request. see also: [what-is-docsearch](https://docsearch.algolia.com/docs/what-is-docsearch/)

The general steps are as follows: 
1. apply,  [docsearch apply](https://docsearch.algolia.com/apply/)
2. waiting for the official reply email.
3. reply and confirm that you are the site owner.
4. wait for the official to reply again, and then configure the information(appId, apikey, indexName) in the email into vitepress theme config. see also: [My configuration](https://github.com/Charles7c/charles7c.github.io/blob/main/docs/.vitepress/config/theme.ts)

![](https://s1.ax1x.com/2022/08/17/vBUQG8.png)

**Reply by MarkusKeck** [reply]
This only applies if your project is open source and no internal project. So for this reason the algolia search should not be the default search option for vitepress. This is such a downgrade from vuepress. You have to share and configure so much just for a simple search.

**Reply by Charles7c** [reply]
Haha, that's true, private project documents are not a good fit.

**Reply by beierzhijin** [reply]
> DocSearch is a service that requires an additional request. see also: [what-is-docsearch](https://docsearch.algolia.com/docs/what-is-docsearch/)
> 
> The general steps are as follows:
> 
> 1. apply,  [docsearch apply](https://docsearch.algolia.com/apply/)
> 2. waiting for the official reply email.
> 3. reply and confirm that you are the site owner.
> 4. wait for the official to reply again, and then configure the information(appId, apikey, indexName) in the email into vitepress theme config. see also: [My configuration](https://github.com/Charles7c/charles7c.github.io/blob/main/docs/.vitepress/config/theme.ts)
> 
> ![](https://camo.githubusercontent.com/244fb92891a5d8c0694533bfc77e53b97856e4dd2aa2ae666b805b5dac5642bb/68747470733a2f2f73312e617831782e636f6d2f323032322f30382f31372f7642555147382e706e67)

我申请之后，给我的回复为啥是这个？回复里的链接还是个404链接😂
![image](https://user-images.githubusercontent.com/54610243/193818297-276ecde6-60bc-4731-bd1e-fa49630f6737.png)

**Comment by Charles7c** [comment]
收到的第一封回信应该是下面这样的，让你确认你是不是你提交的网站的维护者。
![image](https://user-images.githubusercontent.com/25446948/194041540-a9d9a909-3b3a-4847-a79d-0539d5cd7342.png)

**Comment by Charles7c** [comment]
要是最后嫌麻烦，可以考虑用一下本地搜索插件吧。
插件仓库：[vitepress-plugin-search](https://github.com/emersonbottero/vitepress-plugin-search)
我的仓库，克隆下来后只需要把注释打开就可以使用了：https://github.com/Charles7c/charles7c.github.io/blob/main/docs/vite.config.ts

**Comment by beierzhijin** [comment]
谢谢~

就是也不知道为啥 docSearch 在我申请后就给我回了一封邮件，并不是你展示给我的那样

我试了试本地插件，英文搜索还可以 谢谢~

从 Windows 版邮件发送

发件人: Charles Zhang
发送时间: 2022年10月5日 18:44
收件人: vuejs/vitepress
抄送: Beierzhijin; Comment
主题: Re: [vuejs/vitepress] How do I add a search component? The officialDocumentation reads, "Documentation Coming soon..." (Discussion #1154)

要是最后嫌麻烦，可以考虑用一下本地搜索插件吧。
插件仓库：vitepress-plugin-search
我的仓库，克隆下来后只需要把注释打开就可以使用了：https://github.com/Charles7c/charles7c.github.io/blob/main/docs/vite.config.ts
